From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Sat, 21 Jun 2014 16:50:17 +0200
Subject: [121/249] staging: rtl8723au: rtw_joinbss_update_network23a(): update
 beacon info as well
Origin: https://git.kernel.org/linus/2bde0527538802d38b1bed939aefde999600c1ca

We need to copy over the beacon info (interval, capability, tsf) as
well as the IE data.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/core/rtw_mlme.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/staging/rtl8723au/core/rtw_mlme.c b/drivers/staging/rtl8723au/core/rtw_mlme.c
index c91ea8c..b7a3528 100644
--- a/drivers/staging/rtl8723au/core/rtw_mlme.c
+++ b/drivers/staging/rtl8723au/core/rtw_mlme.c
@@ -1035,6 +1035,10 @@ rtw_joinbss_update_network23a(struct rtw_adapter *padapter,
 	memcpy(&cur_network->network.IEs[0], &ptarget_wlan->network.IEs[0],
 	       MAX_IE_SZ);
 
+	cur_network->network.capability = ptarget_wlan->network.capability;
+	cur_network->network.beacon_interval =
+		ptarget_wlan->network.beacon_interval;
+	cur_network->network.tsf = ptarget_wlan->network.tsf;
 	cur_network->aid = pnetwork->join_res;
 
 	rtw_set_signal_stat_timer(&padapter->recvpriv);
